### PR TITLE
added GEH

### PR DESCRIPTION
**Pull request by VasylMoshun**

### CODE REVIEW

File: src/main/java/org/moshun/onlinebookstore/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Positive;
 import java.math.BigDecimal;
 import lombok.Data;
 import lombok.RequiredArgsConstructor;
 
 @Data
 @RequiredArgsConstructor
 public class CreateBookRequestDto {
+    @NotBlank
```
Comment (koval-kn) [inline src/main/java/org/moshun/onlinebookstore/dto/CreateBookRequestDto.java]:
```suggestion
    @NotBlank(message = "Title cannot be null.")
```

File: src/main/java/org/moshun/onlinebookstore/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
     private String title;
+
+    @NotBlank
     private String author;
+
+    @NotBlank
```
Comment (koval-kn) [inline src/main/java/org/moshun/onlinebookstore/dto/CreateBookRequestDto.java]:
```suggestion
    @NotBlank(message = "ISBN cannot be null.")
  @Size(min = 1, max = 13, message = "ISBN must be between 1 and 13 characters.")
```

File: src/main/java/org/moshun/onlinebookstore/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -1,15 +1,24 @@
...
     private String isbn;
+
+    @Positive
     private BigDecimal price;
     private String description;
```
Comment (koval-kn) [inline src/main/java/org/moshun/onlinebookstore/dto/CreateBookRequestDto.java:29]: specify max size for description

File: src/main/java/org/moshun/onlinebookstore/dto/CreateBookRequestDto.java [hunk 4]
```diff
@@ -1,15 +1,24 @@
...
     private String title;
+
+    @NotBlank
     private String author;
+
+    @NotBlank
     private String isbn;
+
+    @Positive
     private BigDecimal price;
```
Comment (koval-kn) [inline src/main/java/org/moshun/onlinebookstore/dto/CreateBookRequestDto.java:26]: not null? specify message

File: src/main/java/org/moshun/onlinebookstore/dto/CreateBookRequestDto.java [hunk 5]
```diff
@@ -1,15 +1,24 @@
...
 import lombok.Data;
 import lombok.RequiredArgsConstructor;
 
 @Data
 @RequiredArgsConstructor
 public class CreateBookRequestDto {
+    @NotBlank
     private String title;
+
+    @NotBlank
```
Comment (koval-kn) [inline src/main/java/org/moshun/onlinebookstore/dto/CreateBookRequestDto.java]: specify message

**Review by koval-kn — CHANGES_REQUESTED** [review]

File: src/main/java/org/moshun/onlinebookstore/dto/CreateBookRequestDto.java [hunk 6]
```diff
@@ -1,16 +1,32 @@
...
 @Data
 @RequiredArgsConstructor
 public class CreateBookRequestDto {
+    @NotBlank(message = "Title cannot be null.")
     private String title;
+
+    @NotBlank(message = "Title cannot be null.")
     private String author;
+
+    @NotBlank
```
Comment (Sanderfuck) [inline src/main/java/org/moshun/onlinebookstore/dto/CreateBookRequestDto.java]: Message for NotBlank?

File: src/main/java/org/moshun/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+            @NotNull HttpHeaders headers,
+            @NotNull HttpStatusCode status,
+            @NotNull WebRequest request) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult().getAllErrors().stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errors);
```
Comment (Sanderfuck) [inline src/main/java/org/moshun/onlinebookstore/exception/CustomGlobalExceptionHandler.java]: Strings can save like constants

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

**Pull request merged**